Source/DafnyCore/Compilers/Compiler-js.cs

```diff
@@ -1317,12 +1341,21 @@ protected class ClassWriter : IClassWriter {
       } else if (e.Value is bool) {
         wr.Write((bool)e.Value ? "true" : "false");
       } else if (e is CharLiteralExpr) {
-        var v = (string)e.Value;
-        wr.Write("'{0}'", v == "\\0" ? "\\u0000" : v);  // JavaScript doesn't have a \0
+        var escaped = TranslateEscapes((string)e.Value);
+        if (UnicodeChars) {
+          wr.Write($"new _dafny.CodePoint('{escaped}'.codePointAt(0))");
+        } else {
+          wr.Write($"'{escaped}'");
+        }
       } else if (e is StringLiteralExpr) {
         var str = (StringLiteralExpr)e;
-        // TODO: the string should be converted to a Dafny seq<char>
-        TrStringLiteral(str, wr);
+        if (UnicodeChars) {
+          wr.Write($"_dafny.Seq.UnicodeFromString(");
```

Comment:
Let's try to give the this the same name in every backend?

Reply:
I'm going to leave that out of scope for now since the backends are already really inconsistent. But integrating the dafny-runtime-in-dafny implementation of `seq<T>` will force more consistency by necessity at least. :)